src/Microsoft.Data/ArrowStringColumn.cs

```diff
@@ -0,0 +1,348 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using Apache.Arrow;
+using Apache.Arrow.Types;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// An immutable to hold Arrow style strings
+    /// </summary>
+    public partial class ArrowStringColumn : BaseColumn
+    {
+        private IList<DataFrameBuffer<byte>> _dataBuffers;
+        private IList<DataFrameBuffer<int>> _offsetsBuffers;
+        private IList<DataFrameBuffer<byte>> _nullBitMapBuffers;
+
+        private readonly Encoding _encoding = Encoding.UTF8;
```

Comment:
Since this is an "Arrow" string, and those are always UTF8, I don't think we need to capture this in a readonly field. We can just use Encoding.UTF8 inline where we need it.